### ISSUE TITLE
Dependency on 'proxy-agent@3.1.0' causes [DEP0005] DeprecationWarning: Buffer()

### ISSUE THREAD

**Issue by baerrach** [bojand/mailgun-js#246]
See https://github.com/TooTallNate/node-proxy-agent/issues/44

**Comment by peterpro** [comment]
and one more: https://github.com/TooTallNate/node-proxy-agent/issues/51

**Comment by onzag** [comment]
Yes I am also getting issues with this package; it's breaking all my https calls.

agent-base is already in version 6...

**Comment by stephent** [comment]
Just walked straight into this one too - since adding mailgun-js, all my other outbound calls using got are broken.

**Comment by JoshuaHintze** [comment]
I noticed this issue to when trying to use https.

Mailgun includes proxy-agent which then includes socks-proxy-agent. 

socks-proxy-agent modifies the https module and does not match the current 14.X LTS release for https.request

which effectively brakes lots of other modules.
